### PR TITLE
Follow hashtags

### PR DESCRIPTION
**Pull request by chdorner**
I had some time this weekend and went ahead and implemented what I wrote down here before getting feedback on Discord. If this is completely wrong then at least it was a good learning opportunity to get to know the codebase :)

There's still an open question on how to implement the `/api/v1/followed_tags` endpoint given that it needs proper IDs for pagination, but the `Hashtag` model just uses the slug as the primary key.

### CODE REVIEW

File: users/models/hashtag_follow.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        return self.get_queryset().by_hashtags(hashtags)
+
+
+class HashtagFollow(models.Model):
+    identity = models.ForeignKey(
+        "users.Identity",
+        on_delete=models.CASCADE,
+        related_name="hashtag_follows",
+    )
+    hashtag = models.ForeignKey(
```
Comment (andrewgodwin) [inline users/models/hashtag_follow.py:26]: This is the target of that query in aget_targets, so I think it should have `db_index=True`

File: api/views/tags.py [hunk 2]
```diff
@@ -14,5 +17,53 @@ def followed_tags(
     min_id: str | None = None,
     limit: int = 100,
 ) -> list[schemas.Tag]:
-    # We don't implement this yet
-    return []
+    queryset = Hashtag.objects.followed_by(request.identity)
+    paginator = MastodonPaginator()
+    # TODO: this fails due to the Hashtag model not having an `id` field
```
Comment (andrewgodwin) [inline api/views/tags.py:22]: Maybe try paginating over the HashtagFollow objects instead?

File: users/models/hashtag_follow.py (hunk 1 above)
Comment (chdorner) [reply users/models/hashtag_follow.py:31]: Added 👍

File: api/views/tags.py (hunk 2 above)
Comment (chdorner) [reply api/views/tags.py]: Yeah, that's a good idea. I added a new `FollowedTag` schema which adds the ID from the `HashtagFollow` model, which works well enough 👍

**Comment by chdorner** [comment]
I left the WIP commit in there for easier re-review but can squash before merge

**Review by andrewgodwin — APPROVED** [review]

**Comment by andrewgodwin** [comment]
I use "squash and merge" on this side anyway, so no need to do it. Thanks for this - a lovely bit of work!

**Pull request merged**